httpx/header.go

```diff
@@ -0,0 +1,18 @@
+package httpx
+
+import "context"
+
+type headerContextKey string
+
+func (c headerContextKey) String() string {
```

Comment:
was this used while debugging things? does it make sense to keep it here or should it be deleted?

Reply:
This is a way to enforce hygiene in the context. Some other library could be writing `cookie` to the context, and so this avoids conflicts.